pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -525,13 +531,12 @@ private boolean buildSegmentInternal(boolean forCommit) {
     // Build a segment from in-memory rows.If buildTgz is true, then build the tar.gz file as well
     // TODO Use an auto-closeable object to delete temp resources.
     File tempSegmentFolder = new File(_resourceTmpDir, "tmp-" + _segmentNameStr + "-" + String.valueOf(now()));
-
+    int nPartitions = _consumerWrapper.getPartitionCount(_kafkaTopic, /*maxWaitTimeMs=*/5000L);
```

Comment:
Not clear what happens if there is a timeout. 

Also, the use cases that do not care to get the number of partitions will experience the delay as well. We have a  budget for segment building now. I would like to see if we can restrict this call to Kafka for use cases that absolutely need it.